hydra/garaga/precompiled_circuits/ec.py

```diff
@@ -196,17 +196,18 @@ def derive_y_from_x(
         y2 = self.field.p - y1  # Negative of y1
 
         # Select y based on s_bit - use larger value if s_bit=1, smaller value if s_bit=0
-        y = y2 if (y1 < y2) == s_bit.value else y1
+        selected_y = y2 if (y1 < y2) == s_bit.value else y1
+        other_y = y1 if (y1 < y2) == s_bit.value else y2
 
         y_coord = self.write_element(
-            y,
+            selected_y,
             WriteOps.WITNESS,
         )
 
         # Validate the y-coordinate
         self.mul_and_assert(y_coord, y_coord, rhs)
 
-        return y_coord
+        return y_coord, self.write_element(other_y, WriteOps.WITNESS)
```

Comment:
Instead of writing two more elements and having s_bit as input, why not just returning y1 and y2 (and not include s_bit as input)

Then in cairo Zero you can find the correct root using s_bit and uint384_is_le

Reply:
Sure, changing.

Reply:
Suggestion was using fp_sqrt (which implictely does mul_and_assert) : 

![image](https://github.com/user-attachments/assets/a6d6833e-a3e7-4156-996d-99f9d034761d)

Reply:
Ok this is much nicer.